nixos/doc/manual/release-notes/rl-2211.section.md

```diff
@@ -92,6 +92,8 @@ In addition to numerous new and upgraded packages, this release has the followin
   as it requires `qt4`, which reached its end-of-life 2015 and will no longer be supported by nixpkgs.
   [According to Barco](https://www.barco.com/de/support/knowledge-base/4380-can-i-use-linux-os-with-clickshare-base-units) many of their base unit models can be used with Google Chrome and the Google Cast extension.
 
+- MongoDB's packages were updated to each latest major version and versions older than 4.0 marked with `meta.known Vulnerabilities` to flag as EOLed by upstream yet still in tree to provide a clean upgrade path, also MongoDB 5.x is the default version from now on.
```

Comment:
```suggestion
- MongoDB's packages were updated to each latest major version and versions older than 4.0 marked with `meta.knownVulnerabilities` to flag as EOLed by upstream yet still in tree to provide a clean upgrade path, also MongoDB 5.x is the default version from now on.
```
that is not quite correct. ``meta.knownVulnerabilities`` has nothing to do with the maintenance status of upstream bit that the package contains long standing not fixed vulnerabilities which is the case because those packages are EOL.

Reply:
This is what we found to mark EOL and give space to users to do a clean upgrade from 3.4 to the latest (be it 5.X or 6.X if https://github.com/NixOS/nixpkgs/pull/190548 gets merged first). 

There is another Nix mechanism to do that? 

And yes we need to keep all versions, as one should upgrade version by version until it reaches the desired one, see https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-standalone/ and https://github.com/NixOS/nixpkgs/issues/155121#issuecomment-1120298542 for reference.

Besides that, I think another way is doing the migration with a container then once the data is OK, start the service in NixOS with the desired version.